mods/farming/api.lua

```diff
@@ -129,62 +112,43 @@ local function tick_again(pos)
 end
 
 -- Seed placement
-farming.place_seed = function(itemstack, placer, pointed_thing, plantname)
-	local pt = pointed_thing
-	-- check if pointing at a node
-	if not pt then
-		return itemstack
-	end
-	if pt.type ~= "node" then
-		return itemstack
-	end
-
-	local under = minetest.get_node(pt.under)
-	local above = minetest.get_node(pt.above)
-
-	local player_name = placer and placer:get_player_name() or ""
-
-	if minetest.is_protected(pt.under, player_name) then
-		minetest.record_protection_violation(pt.under, player_name)
-		return
-	end
-	if minetest.is_protected(pt.above, player_name) then
-		minetest.record_protection_violation(pt.above, player_name)
+function farming.place_seed(itemstack, placer, pt, plantname)
+	if not pt or pt.type ~= "node" or pt.above.y ~= pt.under.y + 1 then
+		-- Seeds can only be placed on top of a node
 		return
```

Comment:
`on_place` must always return the leftover itemstack.
(Also on other places.)

Reply:
fixed